planner/cascades/implementation_rules.go

```diff
@@ -186,3 +192,51 @@ func (r *ImplSelection) OnImplement(expr *memo.GroupExpr, reqProp *property.Phys
 		return nil, plannercore.ErrInternal.GenWithStack("Unsupported EngineType '%s' for Selection.", expr.Group.EngineType.String())
 	}
 }
+
+// ImplSort is the implementation rule which implements LogicalSort
+// to PhysicalSort.
+type ImplSort struct {
+}
+
+// Match implements ImplementationRule match interface.
+func (r *ImplSort) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	ls := expr.ExprNode.(*plannercore.LogicalSort)
+	return plannercore.MatchItems(prop, ls.ByItems)
+}
+
+// OnImplement implements ImplementationRule OnImplement interface.
+func (r *ImplSort) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalProperty) (memo.Implementation, error) {
+	ls := expr.ExprNode.(*plannercore.LogicalSort)
+	ps := plannercore.PhysicalSort{ByItems: ls.ByItems}.Init(
+		ls.SCtx(),
+		expr.Group.Prop.Stats.ScaleByExpectCnt(reqProp.ExpectedCnt),
+		ls.SelectBlockOffset(),
+		&property.PhysicalProperty{ExpectedCnt: math.MaxFloat64},
+	)
+	return impl.NewSortImpl(ps), nil
+}
+
+// ImplNominalSort is the implementation rule which implements LogicalSort
+// to NominalSort.
+type ImplNominalSort struct {
+	newProp *property.PhysicalProperty
+}
+
+// Match implements ImplementationRule match interface.
+func (r *ImplNominalSort) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	ls := expr.ExprNode.(*plannercore.LogicalSort)
+	if !plannercore.MatchItems(prop, ls.ByItems) {
+		return false
+	}
+	newProp, canPass := plannercore.GetPropByOrderByItems(ls.ByItems)
+	newProp.ExpectedCnt = prop.ExpectedCnt
+	r.newProp = newProp
```

Comment:
`r.newProp` would be written concurrently. Move the new property building to `OnImplement`?

Reply:
I also remove the rule `ImplNominalSort`. It seems we don't need to implement both, we only need to choose PhysicaiSort or NominalSort.  If all of the sort by items are columns, we only generate a NominalSort, if the child can not satisfy the order property, the `EnforceRule` will be triggered.